app/services/usps_in_person_proofing/proofer.rb

```diff
@@ -102,17 +102,28 @@ def request_enrollment_code(unique_id)
     end
 
     # Makes a request to retrieve a new OAuth token
-    # and modifies self to store the token and when
+    # and updates the cached token value
     # it expires (15 minutes).
-    # @return [String] the token
+    # @return [String] Auth token
     def retrieve_token!
-      body = request_token
-      @@token_expires_at = Time.zone.now + body['expires_in']
-      @@token = "#{body['token_type']} #{body['access_token']}"
+      token, expires_in, token_type = request_token.fetch_values(
+        'access_token',
+        'expires_in',
+        'token_type',
+      )
+      authz = "#{token_type} #{token}"
+      Rails.cache.write(
+        API_TOKEN_CACHE_KEY, authz,
+        expires_at: Time.zone.now + expires_in
+      )
+      authz
```

Comment:
is there an average/expected expiration for these tokens? Wondering if we could pick a fixed duration and use the block `Rails.cache.fetch(..., expires_in) { }` syntax and simplify a bunch of this code

Reply:
Expected is 15 minutes but it feels weird to hardcode it when we get the actual value in the API response. Agreed it'd be more concise. Mixed feelings about it

Reply:
I would make our version of the token expire slightly _before_ the value provided in the response, maybe by a few seconds.

Reply:
i.e. to reduce the probability that the token will expire between fetching it from cache & attempting to use it.

Reply:
Yeah I had that idea but Andrew convinced me that it's a half-measure and we should just fix it properly by adding in a retry if the token has expired. I created this [ticket](https://cm-jira.usa.gov/browse/LG-7836) for it. Original [PR thread](https://github.com/18F/identity-idp/pull/7147#discussion_r995987682)

Reply:
Why not both?

Reply:
Because it removes the urgency of fixing it better